File: library/src/main/java/com/michaelflisar/recyclerviewpreferences/base/SettingsText.java
package com.michaelflisar.recyclerviewpreferences.base;


import android.widget.TextView;

import com.michaelflisar.recyclerviewpreferences.SettingsManager;

public class SettingsText {

    private int mResText;
    private String mText;

    public SettingsText(int text) {
        mResText = text;
        mText = null;
    }

    public SettingsText(String text) {
        mResText = -1;
        mText = text;
    }

    public void display(TextView tv) {
        if (mResText != -1) {
            tv.setText(mResText);
        } else {
            tv.setText(mText);
        }
    }

    public String getText() {
        if (mResText != -1) {
            return SettingsManager.get().getContext().getString(mResText);
        } else {
            return mText;
        }
    }
}

File: library/src/main/java/com/michaelflisar/recyclerviewpreferences/kt/classes/SettingsText.kt
package com.michaelflisar.recyclerviewpreferences.kt.classes


import android.widget.TextView

import com.michaelflisar.recyclerviewpreferences.SettingsManager

class SettingsText(
        val resText: Int = -1,
        val textString: String? = null
) {
    // TODO: entfernen! nur für java code
    constructor(resText: Int) : this(resText, null)
    constructor(text: String) : this(-1, text)


    val text: String?
        get() = if (resText != -1) {
            SettingsManager.get().context.getString(resText)
        } else {
            textString
        }

    fun display(tv: TextView) {
        if (resText != -1) {
            tv.setText(resText)
        } else {
            tv.text = textString
        }
    }
}
